### PR TITLE
Support a new Messaging API endpoint for friend statistics

### PR DESCRIPTION
**Pull request by okue**
#215

- implement a new function `get_insight_message_event` in `linebot/api.py`
- modify `to_snake_case` in `linebot/utils.py`
  - The snake case of "uniqueMediaPlayed100Percent" should be "unique_media_played_100_percent"

### CODE REVIEW

File: linebot/models/responses.py [hunk 1]
```diff
@@ -392,3 +393,24 @@ def __init__(self, available=None, genders=None, ages=None,
...
+
+    def __init__(self, overview=None, messages=None, clicks=None, **kwargs):
+        """__init__ method.
+
+        :param overview: Percentage per gender.
+        :type overview: T <= :py:class:`linebot.models.StatisticsOverview`
+        :param messages: Percentage per age group.
+        :type messages: list[T <= :py:class:`linebot.models.MessageInfo`]
+        :param clicks: Percentage per area.
+        :type clicks: list[T <= :py:class:`linebot.models.ClickInfo`]
```
Comment (cryeo) [inline linebot/models/responses.py]: [nits]
Please fix the parameter descriptions. It seems like demographics' one.

Comment (cryeo) [reply linebot/models/responses.py]: [imo]
In the case of `InsightDemographicResponse`, response objects have common suffix `Insight`.
If we should follow this naming convention, How about followings?
- `StatisticsOverview` -> `MessageStatistics` or `MessageOverview`
- `MessageInfo` -> `MessageInsight`, `BubbleInsight` or `MessageBubbleInsight` 
- `ClickInfo` -> `ClickInsight`

File: linebot/utils.py [hunk 2]
```diff
@@ -31,8 +31,8 @@ def to_snake_case(text):
     :param str text:
     :rtype: str
     """
-    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', text)
-    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+    s1 = re.sub('(.)([A-Z0-9][a-z]+)', r'\1_\2', text)
```
Comment (cryeo) [inline linebot/utils.py]: [imo]
In the case of this line, `(.)([A-Z][a-z]+)` is enough.

File: linebot/models/responses.py (hunk 1 above)
Comment (okue) [reply linebot/models/responses.py]: Thanks for the good tips.

File: tests/api/test_get_insight.py [hunk 3]
```diff
@@ -187,6 +188,76 @@ def test_get_insight_demographic(self):
...
+        res = self.tested.get_insight_message_event(self.request_id)
+        request = responses.calls[0].request
+        self.assertEqual('GET', request.method)
+        self.assertEqual(
+            request.url,
+            LineBotApi.DEFAULT_API_ENDPOINT +
+            '/v2/bot/insight/message/event?requestId={request_id}'.format(request_id=self.request_id)
+        )
+
+        print(res.overview)
```
Comment (cryeo) [inline tests/api/test_get_insight.py]: [ask]
It seems to be debug code, is it intended?
If not, please remove it.

**Pull request merged**